Add variable explanations (#812)
temporalio/documentation: docs/go/workflows.md

log.Println("Started workflow", "WorkflowID", we.GetID(), "RunID", we.GetRunID())
```

> In the above example, client is imported from "go.temporal.io/sdk/client". 'c' is the temporal client instantiated outside of the snippet. Here we execute a workflow by its type name, namely `MySimpleWorkflow`. By default, the

Reviewer: Thanks for making this clear - though I think we should just rename the variable in the sample to be `temporalClient`:

```
workflowOptions := temporalClient.StartWorkflowOptions{
// ...
}
we, err := temporalClient.ExecuteWorkflow(context.Background(), workflowOptions, FunctionName)
// ...
```

Author: I don't think the above example is valid, because StartWorkflowOptions and Execute workflow are different things. StartWorkflowOptions is on the `go.temporal.io/sdk/client` imported package, and ExecuteWorklow is on the returned struct from `client.NewClient()`.

Reviewer: Ya, you are right, my bad, the options come from the client package - 
So you can ignore that one as an example -
My point is that, any variable is a result of `client.NewClient()` should be named `temporalClient` in our examples.

Author: done!

Reviewer: ```suggestion
In the above example, `client` is imported from the "go.temporal.io/sdk/client" package. And `temporalClient` is an instance of the Temporal Client instantiated outside of the snippet.
Here we spawn a Workflow Execution by its Workflow Type name `MySimpleWorkflow`. By default, the
```
